war/src/main/js/add-item.js

```diff
@@ -0,0 +1,414 @@
+      ]
+    },
+    {
+      order : 1,
+      id : 'folders',
+      display : 'Folders and containers',
+      description : 'Folders and other item types that themselves contain child items.',
+      remainders:'basic',
+      instances : [
+          'com.cloudbees.hudson.plugins.folder.Folder', 
+          'org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject',
+          'jenkins.branch.OrganizationFolder.org.jenkinsci.plugins.github_branch_source.GitHubSCMNavigator',
+          'jenkins.branch.OrganizationFolder.com.cloudbees.jenkins.plugins.bitbucket.BitbucketSCMNavigator'
+      ]
+    },/*
+       * { id:'pipelines', display:'Pipelines and processes',
+       * description:'Complex pipelines and jobs that run through multiple
+       * stages beyond the steps fascilitated by simple jobs.', instances:[
+       * 'org.jenkinsci.plugins.workflow.job.WorkflowJob',
+       * 'com.cloudbees.plugins.flow.BuildFlow',
+       * 'org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject',
+       * 'jenkins.branch.OrganizationFolder.com.cloudbees.jenkins.plugins.bitbucket.BitbucketSCMNavigator',
+       * 'jenkins.branch.OrganizationFolder.org.jenkinsci.plugins.github_branch_source.GitHubSCMNavigator' ] },
+       */
+    {
+      order : 7,
+      id : 'copy',
+      display : 'Copy existing item',
+      placement:'special-single',
+      minToShow:0,
+      description : 'Use an existing item as a starting point for creating a new item configuration.',
+      instances : [
+        'copy'
+      ]
+    },
+    {
+      order : 3,
+      id : 'templates',
+      display : 'Template creators',
+      description : 'Items that allow for the parameerized creation of reusable core system items or components of those items.',
+      remainders:'basic',
+      instances : [
+          'com.cloudbees.hudson.plugins.modeling.impl.jobTemplate.JobTemplate', 
+          'com.cloudbees.hudson.plugins.modeling.impl.folder.FolderTemplate',
+          'com.cloudbees.hudson.plugins.modeling.impl.auxiliary.AuxModel', 
+          'com.cloudbees.hudson.plugins.modeling.impl.builder.BuilderTemplate',
+          'com.cloudbees.hudson.plugins.modeling.impl.publisher.PublisherTemplate'
+
+      ]
+    }, {
+      order : -4,
+      id : 'user-templates',
+      display : 'Custom templates',
+      placement:'special-group',
+      minToShow:1,
+      parents : [
+          {
+            id : 'folderTemplate',
+            display : 'Folder template',
+            description : 'A template for creating parameterized folders',
+            parent : 'com.cloudbees.hudson.plugins.modeling.impl.jobTemplate.JobTemplate'
+          }, {
+            id : 'jobTemplate',
+            display : 'Job template',
+            description : 'A template for creating parameterized jobs',
+            parent : 'com.cloudbees.hudson.plugins.modeling.impl.folder.FolderTemplate'
+          }
+      ],
+      instances : function() {
+      }
+    }
+];
```

Comment:
I'd extract all of this metadata and put it into a `.json` file of its own. You can then do something like...

```javascript
var itemTypes = require('itemTypes.json');
```